### ISSUE TITLE
Support for BINARY collation of CHAR / VARCHAR columns

### ISSUE THREAD

**Issue by VinaiRachakonda** [dolthub/dolt#1488]
now,I can't create tables using the following syntax.

```
CREATE TABLE IF NOT EXISTS `budgets` (
		`id` CHAR ( 36 ) BINARY,
		`uid` INTEGER NOT NULL,
		`name` VARCHAR ( 225 ) NOT NULL UNIQUE,
		`money` INTEGER,
		`created_at` DATETIME NOT NULL,
	  `updated_at` DATETIME  NOT NULL,
	PRIMARY KEY ( `id` )) ENGINE = INNODB COMMENT 'budget' DEFAULT CHARSET = utf8mb4;
```

here is my myql version 

```
mysql  Ver 14.14 Distrib 5.7.32, for osx10.16 (x86_64) using  EditLine wrapper

Connection id:		9
Current database:	
Current user:		root
SSL:			Not in use
Current pager:		less
Using outfile:		''
Using delimiter:	;
Server version:		5.7.9-Vitess 
Protocol version:	10
Connection:		127.0.0.1 via TCP/IP
Server characterset:	
Db     characterset:	
Client characterset:	utf8mb4
Conn.  characterset:	utf8mb4
TCP port:		3306

```

```
mysql> select version();
+-----------+
| VERSION() |
+-----------+
| 8.0.11    |
+-----------+
1 row in set (0.00 sec)
```
this document tell me that binary dataType is supported.
[https://docs.dolthub.com/interfaces/sql/sql-support/data-description](url)
![image](https://user-images.githubusercontent.com/12275804/112712259-2e591600-8f09-11eb-99e2-fe9b68e6865e.png)

_Originally posted by @gorpher in https://github.com/dolthub/dolt/issues/1416#issuecomment-808667001_

**Comment by VinaiRachakonda** [comment]
@zachmu Any idea why this doesn't work?

**Comment by gorpher** [comment]
I found the meaning of this syntax  by looking up the document
```sql
CREATE TABLE IF NOT EXISTS `budgets` (
		`id` CHAR ( 36 ) BINARY, , -- meaning  use 'utf8mb4_bin' Character sorting
	`id_nob` CHAR ( 36 ) , -- meaning  use 'utf8mb4_general_ci' Character sorting
	PRIMARY KEY ( `id` )) ENGINE = INNODB COMMENT 'budget' DEFAULT CHARSET = utf8mb4;
```

**Comment by gorpher** [comment]
I find error start at [sql/parse/parse.go:175 ](https://github.com/dolthub/go-mysql-server/blob/0fef96840e62d5ec3d9fd8e6933925cb08cf8213/sql/parse/parse.go#L175) in go-sql-server project

```go
case *sqlparser.DDL:
		// unlike other statements, DDL statements have loose parsing by default
		ddl, err := sqlparser.ParseStrictDDL(query)
		if err != nil {
			return nil, err
		}
```
sqlparser package in vitess project
It seems that ` binary ` keyword syntax is not supported

**Comment by zachmu** [comment]
Hi @gorpher,

You have the right idea. The `binary` type is not the same as the `char binary` type. The latter is just like `char` but it uses a different collation. Dolt only supports the utf8mb4_general_ci collation right now. Supporting additional collations is on our roadmap, but I don't have a date to give you yet.

In many cases the `binary` type is a fine substitute for `char binary` -- it just stores an inline byte array. Could that work for your use case?

**Comment by gorpher** [comment]
I'm using[ eggjs sequelize](https://sequelize.org/master/manual/model-basics.html) plugin that DataTypes.UUID type field defualt is ` CHAR ( 36 ) BINARY`  

Thank you for your repetition. I hope you can support more collation

**Comment by VinaiRachakonda** [comment]
Skipped bats test here: #4014

**Comment by timsehn** [comment]
Closed in favor of #4019.
